### PR TITLE
Change blueprint schema to YAML

### PR DESCRIPTION
**Pull request by vorostamas**
## Proposed change

Changed the TS blueprint schema to YAML with the possible options.
Feel free to modify it or add suggestions.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/home-assistant.io/pull/15816
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/blueprint/schema.markdown [hunk 1]
```diff
@@ -3,27 +3,88 @@ title: "Blueprint schema"
...
-    input?: Record<string, BlueprintInput | null>;
-    description?: string;
-    source_url?: string;
-  }
-}
+### Schema of the blueprint metadata
+
+<div class='note'>
+
+Values inside `<>` are required, inside `[]` are optional.
```
Comment (spacegaier) [inline source/_docs/blueprint/schema.markdown:10]:
```suggestion
Values inside `<>` are required, inside `[]` are optional. Values separated by `|` represent options (`[A | B]` means either `A` or `B`).
```

File: source/_docs/blueprint/schema.markdown [hunk 2]
```diff
@@ -3,27 +3,88 @@ title: "Blueprint schema"
...
+
+</div>
+
+Blueprint object:
+```yaml
+blueprint:
+  name: <string>
+  description: <string>
```
Comment (spacegaier) [inline source/_docs/blueprint/schema.markdown:18]:
```suggestion
  description: [string]
```

It's optional though encouraged to be used.

File: source/_docs/blueprint/schema.markdown [hunk 3]
```diff
@@ -3,27 +3,88 @@ title: "Blueprint schema"
...
+
+Blueprint Input:
+```yaml
+name: [string]
+description: [string]
+selector: [Selector]
+default: [any]
+```
+
+Selectors:
```
Comment (balloob) [inline source/_docs/blueprint/schema.markdown:32]: This should move to the selectors page. Selectors are not Blueprint specific.

File: source/_docs/blueprint/schema.markdown (hunk 1 above)
Comment (frenck) [inline source/_docs/blueprint/schema.markdown:10]: We should not all of a sudden introduce new standards.

We have developer documentation on how to document YAML and defined standards for placeholders. We also have helpers for showing YAML properties, their types, if they are optional or not and provide a description for each of those.

If anything, we should stick to a standard. I'm happy to improve standards, but that should be done site wide.

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
I've replaced this PR with #15873

Thanks for the input and inspiration to improve this page @vorostamas 👍

File: source/_docs/blueprint/schema.markdown (hunk 1 above)
Comment (vorostamas) [reply source/_docs/blueprint/schema.markdown:10]: I can fully agree with you, thank you for looking into this.

**Pull request closed without merging**